iree/base/internal/wait_handle_test.cc

```diff
@@ -211,10 +211,10 @@ TEST(WaitSet, Lifetime) {
 
 TEST(WaitSet, UnreasonableCapacity) {
   iree_wait_set_t* wait_set = NULL;
-  IREE_EXPECT_STATUS_IS(
-      IREE_STATUS_INVALID_ARGUMENT,
-      iree_wait_set_allocate(1 * 1024 * 1024, iree_allocator_system(),
-                             &wait_set));
+  iree_status_t status = iree_wait_set_allocate(
+      1 * 1024 * 1024, iree_allocator_system(), &wait_set);
+  IREE_EXPECT_STATUS_IS(IREE_STATUS_INVALID_ARGUMENT, status);
```

Comment:
Do we have any cases where we don't want to free a status after `IREE_EXPECT_STATUS_IS`? Could create a `IREE_EXPECT_STATUS_IS_AND_FREE` :P

Reply:
Yeah I thought about that. The other thing I was thinking was that it should be possible to free rvalues only. I didn't go to deep into it though, since this was simple and Ben was saying that he was going to do something to make this better with a C++ RAII wrapper or something